.github/workflows/pr.yml

```diff
@@ -0,0 +1,52 @@
+name: CI
+
+# Controls when the action runs.
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  generate:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      # Setup go and install dependencies.
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.16
+
+      # See https://grpc.io/docs/protoc-installation/#install-pre-compiled-binaries-any-os
+      - name: Install Protocol Buffer Compiler
+        run: |
+          PB_REL="https://github.com/protocolbuffers/protobuf/releases"
+          curl -LO $PB_REL/download/v3.15.8/protoc-3.15.8-linux-x86_64.zip
+          unzip protoc-3.15.8-linux-x86_64.zip -d $HOME/.local
+          export PATH="$PATH:$HOME/.local/bin"
+
+      - name: Set up Go dependencies
+        run: |
+          go version
+          go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.26
+          go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.1
+          echo $PATH
+
+      # Run our own verifications and scripts.
+      - name: Verify that there is no diff after running go mod tidy
+        run: go mod tidy && git status && git diff-index --quiet HEAD --
+
+      - name: Verify that there is no diff after running go generate
+        run: go generate ./... && git status && git diff-index --quiet HEAD --
```

Comment:
We might want to have an new line at the end of the file.

Generally, the best practice stems from the use of `cat` to output file contents on the terminal. Without the final new line, you end up with your prompt after the last line, instead of on its own new line.

Reply:
Will fix, thanks! (I know about the issue, happens every now and then with my IDE, I should have double-checked the diff 👀)